### PR TITLE
feature: login is bug fixed

### PR DESCRIPTION
**Pull request by LONGLIN163**
let's talk if the solution is ok or not...

### CODE REVIEW

**Review by dev-long-lin — COMMENTED** [review]
Almost perfect if you can add clear comments.

File: file3.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+hello
```
Comment (dev-long-lin) [inline file3.txt]: Please try do not use non-official library.

**Review by dev-long-lin — CHANGES_REQUESTED** [review]
please make this change and do another push.

File: file3.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+clear comment added
```
Comment (dev-long-lin) [inline file3.txt]: nice

**Review by dev-long-lin — APPROVED** [review]
well done

**Review by dev-long-lin — COMMENTED** [review]
ok

File: file3.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+email_info_validation
```
Comment (test-long-lin) [inline file3.txt]: Please add more detail assertion.

**Review by test-long-lin — CHANGES_REQUESTED** [review]
Add more assertion:
string length
combining-character with certain types

File: file3.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+start test with three member
```
Comment (dev-long-lin) [inline file3.txt]: ok

**Review by dev-long-lin — COMMENTED** [review]
ok I am the one who push this commits

File: file3.txt (hunk 4 above)
Comment (test-long-lin) [reply file3.txt]: Got it

**Review by test-long-lin — COMMENTED** [review]
I am the one who will check your code.

**Review by test-long-lin — APPROVED** [review]

File: file3.txt (hunk 4 above)
Comment (dev-long-lin) [reply file3.txt]: ok2

**Review by dev-long-lin — COMMENTED** [review]
ok2...

File: file3.txt (hunk 4 above)
Comment (dev-long-lin) [reply file3.txt]: I change login logic with decorator pattern...

**Review by dev-long-lin — COMMENTED** [review]
I update login logic code

File: file3.txt (hunk 4 above)
Comment (test-long-lin) [reply file3.txt]: ok I know, I will update test snipets.

**Review by test-long-lin — COMMENTED** [review]
updating test code

File: file3.txt (hunk 4 above)
Comment (dev-long-lin) [reply file3.txt]: have you update

Comment (test-long-lin) [reply file3.txt]: yes

Comment (test-long-lin) [reply file3.txt]: have you change again

Comment (dev-long-lin) [reply file3.txt]: no i would not change it anymore

**Pull request merged**